### PR TITLE
tplgtool2: compatible with construct 2.8-2.10

### PR DESCRIPTION
**Pull request by miRoox**
Conditional import to ensure tplgtool2 compatible with construct 2.8-2.10.
To migrate Ubuntu 20.04 (construct 2.8.16) to 22.04 (construct 2.10.67).

see also:
- https://packages.ubuntu.com/jammy/python3-construct
- https://github.com/construct/construct/blob/master/docs/transition29.rst
- https://github.com/construct/construct/blob/master/docs/transition210.rst

Signed-off-by: Yongan Lu <yongan.lu@intel.com>

### CODE REVIEW

File: tools/tplgtool2.py [hunk 1]
```diff
@@ -8,19 +8,25 @@
...
+    from construct import String
+
+    def __enum_dict(enumtype):
+        return dict((e.name, e.value) for e in enumtype)
+    # here is a workaround to make Enum/FlagsEnum accept enum type, like construct 2.9
+    def Enum(subcon, enumtype):
+        return construct.Enum(subcon, default=Pass, **__enum_dict(enumtype))
+    def FlagsEnum(subcon, enumtype):
+        return construct.FlagsEnum(subcon, **__enum_dict(enumtype))
+if (2, 9, 0) <= construct.version:
```
Comment (marc-hb) [inline tools/tplgtool2.py:27]: skip a line before this?

File: tools/tplgtool2.py [hunk 2]
```diff
@@ -8,19 +8,25 @@
...
-    return dict((e.name, e.value) for e in enumtype)
-
-# here is a workaround to make Enum/FlagsEnum accept enum type, like construct 2.9
-def Enum(subcon, enumtype):
-    return construct.Enum(subcon, default=Pass, **__enum_dict(enumtype))
-
-def FlagsEnum(subcon, enumtype):
-    return construct.FlagsEnum(subcon, **__enum_dict(enumtype))
+# Pylint complain about the missing names even in conditional import.
+# pylint: disable=E0611
```
Comment (marc-hb) [inline tools/tplgtool2.py:16]: Can you move this useful link from the commit message to a comment?
https://github.com/construct/construct/blob/master/docs/transition29.rst

**Review by marc-hb — COMMENTED** [review]
Why block < 2.8.0 now? Only because it's not tested? It wasn't blocked before.

Whether blocking older versions is the right decision or not, I don't like adding this block AND support for newer version _in the same commit_.

File: tools/tplgtool2.py (hunk 2 above)
Comment (miRoox) [reply tools/tplgtool2.py:16]: I append it after the condition.

**Comment by miRoox** [comment]
Tested http://sof-ci.sh.intel.com/#/result/planresultdetail/12505?model=APL_UP2_NOCODEC&testcase=verify-tplg-binary, it works on Ubuntu 22.04.

**Comment by marc-hb** [comment]
SOFCI TEST

**Comment by marc-hb** [comment]
SOFCI TEST

**Review by aiChaoSONG — APPROVED** [review]

**Comment by aiChaoSONG** [comment]
merge for the compatibility with ubuntu 22.04

**Pull request merged**